vivo-project/Vitro
# Feature/forgot password

[VIVO PR](https://github.com/vivo-project/VIVO/pull/3916)
This PR solves ["I forgot my pasword" issue](https://github.com/vivo-project/VIVO/issues/3770).

# What does this pull request do?
This PR allows the user to request password change from login widget, instead of having to ask an admin to do it for him.

# What's new?
Password reset functionality with built-in anti-spam measures. I have added a new servlet (ForgotPassword) that handles the logic. I have also updated i18n files with required frontend labels. Have in mind that I only speak English and Serbian (and a tiny bit of German :D) so I am not well suited for translation. For majority of languages I have used Google Translate so if you notice any translation errors please leave a comment on this PR.

# How should this be tested?
Try to change your password through login widget. Try to spam and see what happens.

# Additional Notes:
Removed old captcha implementation files which was replaced in [PR](https://github.com/vivo-project/Vitro/pull/427/files) 
jquery.realperson.css
jquery.realperson.js

# Admin notification email translations:

## English
![image](https://github.com/vivo-project/Vitro/assets/57297965/f076c269-f642-4c72-b79d-302c47827e63)

## German
![image](https://github.com/vivo-project/Vitro/assets/57297965/dce5d877-daaa-43a4-9852-624456ac0b97)

## Spanish
![image](https://github.com/vivo-project/Vitro/assets/57297965/a494a439-5046-45fc-9587-72ef5e5f56c8)

## Russian
![image](https://github.com/vivo-project/Vitro/assets/57297965/3249e181-ae00-4e4f-b126-4b4b9c9163f7)

## Portuguese
![image](https://github.com/vivo-project/Vitro/assets/57297965/76d15421-b348-4bc8-88e4-f5e179d1f0ec)

## French-Canadian
![image](https://github.com/vivo-project/Vitro/assets/57297965/a818a1a6-aa52-4ea5-8e64-198c28656426)

# Interested parties
@chenejac @litvinovg @brianjlowe
